src/shadow-embed.js

```diff
@@ -548,6 +548,31 @@ export class ShadowDomWriterStreamer {
     return this.success_;
   }
 
+  /** @override */
+  get closed() {
+    return this.eof_;
+  }
+
+  /** @override */
+  get desiredSize() {
+    return 1; // noop
```

Comment:
I read the spec and it says that when it is below 0 it is over-full and I assume buffering but not actually writing to the stream, and when it is 0 it is closed.
https://streams.spec.whatwg.org/#default-writer-desiredSize

Thinking about it now, a better value could be `return this.eof_ ? 0 : 1;`